### ISSUE TITLE
Blank screen and JavaScript syntax errors in IE11 when using Material theme

### ISSUE THREAD

**Issue by pleku** [vaadin/vaadin-material-styles#85]
@evmatt commented on [Wed Feb 27 2019](https://github.com/vaadin/flow/issues/5140)

This bug appears to be similar to #4944. When running an application in production mode on Vaadin 12.0.7, which [should incorporate ](https://github.com/vaadin/flow/releases/tag/1.2.5) the fix for #4944, the user sees a blank screen in IE11. Three JavaScript SyntaxErrors appear in the console on loading the page, all from the file `style-properties.js`. On inspecting this file ES6 keywords such as `const` are seen, so it appears that there is a failure in transpilation from ES6 to ES5 during the build process. This only occurs when using the Material theme, the Lumo theme works as expected.

I have been able to reproduce this bug using the Beverage Buddy starter pack running 12.0.7 when using the Material theme. When running the starter pack the errors arise from the `webcomponents-lite.js` file rather than `style-properties.js`.

---

@ZheSun88 commented on [Thu Feb 28 2019](https://github.com/vaadin/flow/issues/5140#issuecomment-468230658)

Issue confirmed.
Bug can be reproduced by running production build on beverage-start using `material` theme, though the js error is different every time i run, for example one from `webcomponents-lite.js`, which comes from `webcomponentsjs` webjar.
![image](https://user-images.githubusercontent.com/31067185/53562886-393ef200-3b5b-11e9-87b4-2ec3e821fcf1.png)

**Comment by Haprog** [comment]
I was able to reproduce the issue just by running the `vaadin-material-styles` demo with `polymer serve` and accessing it in IE11. I was testing in the `1.x` branch which is the same as v1.2.2 atm. but it was difficult to get it reproduced since this version has another bug (which makes the demo use wrong fonts in Edge and IE11) which was somehow preventing this blank page issue and JS errors from happening, but once I fixed the font issue locally, then this JS errors and blank page started happening.

I have a working fix coming up (just needs a review and some releases).

**Comment by Haprog** [comment]
This fix has been released in `vaadin-material-styles` [v1.2.3](https://github.com/vaadin/vaadin-material-styles/releases/tag/v1.2.3)

The problem was not transpilation (which has been working correctly all the time). You might be shown the ES6 sources in dev tools (even in IE11) due to source maps, but that was not the issue.

The problem was actually "just CSS". See [the commit message](https://github.com/vaadin/vaadin-material-styles/commit/5fe9f9ebbf2aeaf3532ad25b7b226af5911683ca) for more details.

I also reproduced the issue (and verified the fix) using Beverage Buddy demo with Vaadin version 13.0.7 to confirm that the fix also works in a Java/Flow project.

This fix should be included in the next maintenance release for Vaadin 13, but until then you can use this fix in a Vaadin Java project by adding this explicit version dependency into your `pom.xml`:
```xml
<dependency>
    <groupId>org.webjars.bowergithub.vaadin</groupId>
    <artifactId>vaadin-material-styles</artifactId>
    <version>1.2.3</version>
</dependency>
```

Note: If you change Beverage Buddy demo to use the material theme, in addition to adding `@Theme(Material.class)` to `MainLayout.java` you should also edit `shared-styles.html` so that it doesn't try to import Lumo styles and doesn't depend on the `--lumo-*` variables.
